### PR TITLE
Delete multiple tasks

### LINKED ISSUE #43: Delete multiple tasks at once
As a university student, I can delete multiple tasks at once, so that I can clean up my tasks in the app easily

### PR DESCRIPTION
**Pull request by CSmortal**
Closes #43

### CODE REVIEW

**Review by khiaxeng — APPROVED** [review]
LGTM, good job on the testcases!

File: src/main/java/seedu/taskify/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -34,9 +43,52 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
...
+     * @throws ParseException if {@code oneBasedIndexes} cannot be parsed properly
+     */
+    public static List<Index> parseMultipleIndex(String oneBasedIndexes) throws ParseException {
+        assert hasMultipleValidIndex(oneBasedIndexes) : "ParserUtil#parseMultipleIndex should not be called on this "
+                + "string";
+        String[] arguments = extractStringArgumentsIntoIndexes(oneBasedIndexes);
+        boolean hasOnlyOneArgument = arguments.length == 1;
+
+        if (hasOnlyOneArgument) {
+            // throw new AssertionError instead?
```
Comment (noelmathewisaac) [inline src/main/java/seedu/taskify/logic/parser/ParserUtil.java:63]: I think exceptions are more suitable as we are trying to figure out errors with user input

File: src/main/java/seedu/taskify/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -34,9 +43,52 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
...
     }
 
+    /**
+     * Parses {@code oneBasedIndexes} into a list of {@code Index} and returns it.
+     * @param oneBasedIndexes user's input excluding the command word
+     * @return a list of {@code Index} representing all valid indexes in {@code oneBasedIndexes}
+     * @throws ParseException if {@code oneBasedIndexes} cannot be parsed properly
+     */
+    public static List<Index> parseMultipleIndex(String oneBasedIndexes) throws ParseException {
+        assert hasMultipleValidIndex(oneBasedIndexes) : "ParserUtil#parseMultipleIndex should not be called on this "
```
Comment (noelmathewisaac) [inline src/main/java/seedu/taskify/logic/parser/ParserUtil.java:57]: Should an exception be used here instead?

**Review by noelmathewisaac — COMMENTED** [review]
Few clarifications about assertions vs exceptions, otherwise great job!

File: src/main/java/seedu/taskify/logic/parser/ParserUtil.java (hunk 2 above)
Comment (CSmortal) [reply src/main/java/seedu/taskify/logic/parser/ParserUtil.java:57]: ParseExceptions are thrown in line 59 and line 64 for unexpected user input, my assertion in line 57 was mainly to warn whoever make changes to our code against using this public method. It should only be called if the user's argument input passes `#hasMultipleValidIndex` first, as seen in line 31 of `DeleteMultipleCommandParser` and line 59 of `TaskifyParser`.

File: src/main/java/seedu/taskify/logic/parser/ParserUtil.java (hunk 1 above)
Comment (CSmortal) [reply src/main/java/seedu/taskify/logic/parser/ParserUtil.java:63]: Sorry, can I clarify which line you are referring to?

Comment (noelmathewisaac) [reply src/main/java/seedu/taskify/logic/parser/ParserUtil.java:63]: I was commenting on your comment on line 63 haha

File: src/main/java/seedu/taskify/logic/parser/ParserUtil.java (hunk 2 above)
Comment (noelmathewisaac) [reply src/main/java/seedu/taskify/logic/parser/ParserUtil.java:57]: Ok makes sense

**Review by noelmathewisaac — APPROVED** [review]

**Pull request merged**